apollo-client/src/main/java/io/logz/apollo/clients/ApolloClient.java

```diff
@@ -99,23 +99,32 @@ public DeployableVersion getDeployableVersionFromSha(String sha, int serviceId)
         return genericApolloClient.getResult("/deployable-version/latest/service/" + serviceId, new TypeReference<List<DeployableVersion>>() {});
     }
 
-    public Deployment addDeployment(Deployment deployment) throws ApolloClientException {
-        String requestBody = Common.generateJson("environmentId", String.valueOf(deployment.getEnvironmentId()),
-                "serviceId", String.valueOf(deployment.getServiceId()),
+    public MultiDeploymentResponseObject addDeployment(Deployment deployment) throws ApolloClientException {
+        String requestBody = Common.generateJson("environmentIdsCsv", String.valueOf(deployment.getEnvironmentId()),
+                "serviceIdsCsv", String.valueOf(deployment.getServiceId()),
                 "deployableVersionId", String.valueOf(deployment.getDeployableVersionId()),
                 "deploymentMessage", deployment.getDeploymentMessage());
 
-        return genericApolloClient.postAndGetResult("/deployment", requestBody, new TypeReference<Deployment>() {});
+        return genericApolloClient.postAndGetResult("/deployment", requestBody, new TypeReference<MultiDeploymentResponseObject>() {});
+    }
+
+    public MultiDeploymentResponseObject addDeploymentWithParams(String environmentIdsCsv, String serviceIdsCsv, int deployableVersionId) throws ApolloClientException {
```

Comment:
why the "withParams" part? just another overload

Reply:
And please just construct a deployment object here and pass to the first overload, or the other way around. anyway no use for this duplication